kolena/_experimental/object_detection/dataset.py

```diff
@@ -305,15 +311,28 @@ def _compute_metrics(
         )
 
 
+def _safe_get_label(obj: object) -> Optional[str]:
+    if hasattr(obj, "label"):
+        return str(obj.label)
+
+    return None
+
+
+def _get_labels_from_objects(objs: Iterable[object]) -> List[str]:
+    maybe_labels = {_safe_get_label(obj) for obj in objs}
+    labels = [label for label in maybe_labels if label is not None]
+    return labels
+
+
 def _check_multiclass(ground_truth: pd.Series, inference: pd.Series) -> bool:
-    try:
-        labels = {x.label for x in itertools.chain.from_iterable(_filter_null(ground_truth))}.union(
-            {x.label for x in itertools.chain.from_iterable(_filter_null(inference))},
-        )
-        return len(labels) >= 2
-    except AttributeError:
+    labels = {_safe_get_label(gt) for gt in itertools.chain.from_iterable(_filter_null(ground_truth))}.union(
+        {_safe_get_label(inf) for inf in itertools.chain.from_iterable(_filter_null(inference))},
+    )
+    if None in labels:
         return False
 
+    return len(labels) >= 2
+
```

Comment:
I personally think this logic is wrong. I think we should ignore objects without labels and check whether there is more than 1 unique label, instead of just giving up and defaulting to single class when there is an object without a label.

I kept the existing behavior for now because I'm not sure what consequences changing this behavior has for downstream handling of thresholded metrics.

Reply:
I don't think there is a case where we mix unlabeled vs labeled annotations for OD, which is why it short circuits like this -- I agree that we should ideally check all valid labels, rather than declare that it's a single class problem.